distributed/process.py

```diff
@@ -81,15 +81,26 @@ def __init__(self, loop=None, target=None, name=None, args=(), kwargs={}):
                 dask.config.global_config,
             ),
         )
-        _dangling.add(self._process)
         self._name = self._process.name
+        self._proc_finalizer = weakref.finalize(self, self._get_finalizer())
         self._watch_q = PyQueue()
         self._exit_future = Future()
         self._exit_callback = None
         self._closed = False
 
         self._start_threads()
 
+    def _get_finalizer(self):
+        def finalize(proc=self._process, r=repr(self)):
```

Comment:
`r` is never referenced in the function, could be removed

Reply:
Good catch, perks of using copy/paste. :) 

Fixed in https://github.com/dask/distributed/pull/4184/commits/d61f92932a9aa775d436f05125f5096a8cd2cf1a .